note

```diff
@@ -32,62 +41,67 @@ Arguments:
   -e | --edit <DATE: year-month-day>  Open a specific note for editing.
   -p | --print                        Print the contents of a note.
   -c | --create                       Create a note but don't open it for editing.
-  -i | --info                         Print information about a note.\n"
+  -n | --name                         Set filename for note. Will be created in \$NOTE_DIR
+                                      Don't forget an extension like .md
+  -i | --info                         Print information about a note.
+
+The script loads configuration variables from \$XDG_CONFIG_HOME/note/noterc.
+
+Example:
+# Directory where the current note should be stored
+NOTE_DIR=\"\$HOME/notes/\$YEAR/\$MONTH\"
+
+# Name of the Note
+NOTE_NAME=\"\$DAY.md\"\n"
 }
 
-open_note () {
+open_note() {
     if [[ $EDITOR = *"vim"* ]] || [[ $EDITOR = *"nvim"* ]]; then
         # Open Vim or Neovim in insert mode.
-        $EDITOR "+normal G$" +startinsert! $note_path
+        $EDITOR "+normal G$" +startinsert! "$NOTE_DIR/$NOTE_NAME"
     elif [[ $EDITOR = *"emacs"* ]]; then
         # Open Emacs with cursor at EOF.
-        emacs -nw $note_path --eval "(goto-char (point-max))"
+        emacs -nw "$NOTE_DIR/$NOTE_NAME" --eval "(goto-char (point-max))"
     elif [[ $EDITOR = "" ]]; then
         # If no default editor, use Vim and open in insert mode.
-        vim "+normal G$" +startinsert! $note_path
+        vim "+normal G$" +startinsert! "$NOTE_DIR/$NOTE_NAME"
     else
-        $EDITOR $note_path
+        $EDITOR "$NOTE_DIR/$NOTE_NAME"
     fi
 }
 
-year=`date +'%Y'`
-month=`date +'%m'`
-day=`date +'%d'`
-note_path=~/notes/$year/$month/$day.md
-
-if (( $# > 0 )); then
-    while [[ $# -gt 0 ]]
-    do
+if (($# > 0)); then
+    while [[ $# -gt 0 ]]; do
         key="$1"
         case $key in
-            -e|--edit)
+        -e | --edit)
             printf "(e)dit is not yet implemented :(\n"
-            #shift
-            #shift
             exit 0
             ;;
-            -p|--print)
-            cat $note_path
-            shift
+        -p | --print)
+            cat "$NOTE_DIR/$NOTE_NAME"
```

Comment:
Should probably add an optional argument for note name if we're allowing custom named notes via `-n`
e.g.: `note -p mynote.md` or  `note -p -n mynote.md`

Reply:
Now it should be possible to use -n in combination with other options. If no other option present, it opens the note. Not the best way how I did it but didn't came up with a better solution

Reply:
It does work, but the `-n name` option must be first. That's OK for now, i'll look at upgrading that in the future.